src/realm/sync/changeset_encoder.cpp

```diff
@@ -224,12 +224,14 @@ void ChangesetEncoder::operator()(const Instruction::SetErase& instr)
 
 InternString ChangesetEncoder::intern_string(StringData str)
 {
-    auto it = m_intern_strings_rev.find(str);
+    auto it = m_intern_strings_rev.find(static_cast<std::string_view>(str));
     if (it == m_intern_strings_rev.end()) {
         size_t index = m_intern_strings_rev.size();
         // FIXME: Assert might be able to be removed after refactoring of changeset_parser types?
         REALM_ASSERT_RELEASE_EX(index <= std::numeric_limits<uint32_t>::max(), index);
-        it = m_intern_strings_rev.insert({std::string{str}, uint32_t(index)}).first;
+        bool inserted;
+        std::tie(it, inserted) = m_intern_strings_rev.insert({std::string{str}, uint32_t(index)});
```

Comment:
In other places we just use `auto [it, inserted]` for structured bindings.

Reply:
structured bindings won't work if one of the bindings has already been defined as a variable. `std::tie()` works around that.

Reply:
Of course.